src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Badge.ts

```diff
@@ -179,74 +196,44 @@ export var badgeAssignmentCreateDirective = (
         link: (scope, element) => {
             bindPath(adhHttp, adhPermissions, $q)(scope);
 
-            scope.submit = () => {
-                var postdata = {
-                    content_type: RIBadgeAssignment.content_type,
-                    data: {}
-                };
-                return adhHttp.post(scope.poolPath, fill(postdata, scope, adhCredentials.userPath))
-                    .then((response) => {
-                        scope.serverError = null;
-                        if (scope.onSubmit) {
-                            scope.onSubmit();
-                        }
-                    }, (response) => {
-                        scope.serverError = response[0].description;
-                    });
-            };
-
-            scope.cancel = () => {
-                if (scope.onCancel) {
-                    scope.onCancel();
-                }
-            };
-        }
-    };
-};
+            adhHttp.get(scope.badgeablePath).then((proposal) => {
+                scope.poolPath = proposal.data[SIBadgeable.nick].post_pool;
 
-export var badgeAssignmentEditDirective = (
-    adhConfig : AdhConfig.IService,
-    adhHttp : AdhHttp.Service<any>,
-    adhPermissions: AdhPermissions.Service,
-    $q : angular.IQService,
-    adhCredentials : AdhCredentials.Service
-) => {
-    return {
-        restrict: "E",
-        templateUrl: adhConfig.pkg_path + pkgLocation + "/Assignment.html",
-        scope: {
-            path: "@",
-            poolPath: "@",
-            showDescription: "=?",
-            onSubmit: "=?",
-            onCancel: "=?"
-        },
-        link: (scope, element) => {
-            bindPath(adhHttp, adhPermissions, $q)(scope, "path");
-
-            scope.delete = () => {
-                return adhHttp.delete(scope.path, RIBadgeAssignment.content_type)
-                    .then(() => {
-                        scope.serverError = null;
-                        if (scope.onSubmit) {
-                            scope.onSubmit();
-                        }
-                    }, (response) => {
-                        scope.serverError = response[0].description;
-                    });
-            };
+                return adhGetBadges(proposal).then((assignments : IBadge[]) => {
+                    scope.assignments = _.keyBy(assignments, "badgePath");
+                    scope.checkboxes = _.mapValues(scope.assignments, (v) => true);
```

Comment:
I don't understand the second argument yet: all assignments are true? But how do the other checkboxes get created?